geoh5py/objects/object_base.py

```diff
@@ -177,13 +178,24 @@ def add_data(
 
         property_groups: dict[PropertyGroup | None, list[Data]] = {}
         data_objects = []
+
+        names = [
+            child.name
+            for child in self.children
+            if isinstance(child, Data) and child.name != "Visual Parameters"
+        ]
+
         for name, attr in data.items():
             if not isinstance(attr, dict):
                 raise TypeError(
                     f"Given value to data {name} should of type {dict}. "
                     f"Type {type(attr)} given instead."
                 )
 
+            name = find_unique_name(name, names)
```

Comment:
We probably want to prevent VisualParameters all together. Here it could add another `Visual Parameters` with the same name.

Reply:
So right now, the way it's written
1- You pull all names other than Visual Parameters
2- Loop over all entries and increment the names (if needed). If the user passes a "Visual Parameter" data, then the name won't be incremented and potentially end up with two "Visual Parameters" if one exists already, which is a problem.

My point was to simply raise a `UserWarning` right away if the name "Visual Parameter" is encountered in the `add_data`. It should never be encountered here in the first place. Either it is coming from GA, or it is forced to go through the `add_default_visual_parameters` on the object to force formatting. And you can remove the check below as well.